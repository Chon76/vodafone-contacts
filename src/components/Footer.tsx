import { Text } from 'native-base';

export default function Footer() {
  return (
    <Text my={6} fontWeight={'bold'} textAlign={'center'}>
      All Rights Reserved _VOIS 2022
    </Text>
  );
}
