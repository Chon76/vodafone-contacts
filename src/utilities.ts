import { Contact } from './types/Contact';

// eslint-disable-next-line import/prefer-default-export
export function getInitialLeters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  console.log('initials', initialLetters);

  return initialLetters;
}
