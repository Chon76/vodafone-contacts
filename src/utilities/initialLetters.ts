import { Contact } from '../types/Contact';

/**
 *
 * @param contacts an array of {@link Contact}
 * @returns the initial letters of the names of the contact
 */
// eslint-disable-next-line import/prefer-default-export
export function getInitialLetters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  return initialLetters;
}
