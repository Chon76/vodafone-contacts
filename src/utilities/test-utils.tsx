/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NativeBaseProvider } from 'native-base';
import { NavigationContainer } from '@react-navigation/native';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const wrapper = ({ children }: { children: React.ReactNode }) => {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  );
};

// wrap tests with necessery providers
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper, ...options });

export * from '@react-navigation/native';
export { customRender as render };
