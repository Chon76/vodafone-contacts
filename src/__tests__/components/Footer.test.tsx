import Footer from '../../components/Footer';

import { render } from '../../utilities/test-utils';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('footer', () => {
  it('renders corretly ', () => {
    const footer = render(<Footer />);

    expect(footer.toJSON()).toMatchSnapshot();
  });
});
