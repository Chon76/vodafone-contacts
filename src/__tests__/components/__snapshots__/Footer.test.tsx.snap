// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders corretly  1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    dataSet={{}}
    style={
      {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 24,
        "marginBottom": 24,
        "marginTop": 24,
        "textAlign": "center",
        "textDecorationLine": undefined,
      }
    }
  >
    All Rights Reserved _VOIS 2022
  </Text>
</RNCSafeAreaProvider>
`;
